internal/upload/upload.go

```diff
@@ -141,7 +150,20 @@ func NewHandler(
 		}
 
 		if isTestRequest(r) {
+			mockVr := &validators.Request{
```

Comment:
What do you think about moving this logic into a function (something like `handleTestRequest()` etc)?  That would help shorten up this function (which is getting pretty long) and start lessening the cognitive load required to read this function.

It would be nice if we could get this main function to be an outline of the request handling logic.  This function would defer to other functions to add the lower level logic.  I hope that makes sense.

Just a thought...